### PR TITLE
added blog example

### PR DESCRIPTION
**Pull request by harshmangalam**

### CODE REVIEW

File: examples/blog/blog/build-your-own-github-using-deno-fresh-and-github-rest-api-32h6.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      const username = formData.get("username");
+      // add validation
+      if (!username || username.toString().trim().length === 0) {
+        return ctx.render({ error: "Username should not be empty" });
+      }
+
+      // check if username exists
+      const [status] = await fetchUserInfo(String(username));
+      // handle different error status code
+      if (status === 404) {
```
Comment (thegaryroberts) [inline examples/blog/blog/build-your-own-github-using-deno-fresh-and-github-rest-api-32h6.md]: As per this PR: https://github.com/denoland/fresh/pull/417 you could change these values [(magic numbers)](https://en.wikipedia.org/wiki/Magic_number_(programming)) to use the standard HTTP status codes: https://doc.deno.land/https://deno.land/std@0.146.0/http/http_status.ts/~/Status

Comment (harshmangalam) [reply examples/blog/blog/build-your-own-github-using-deno-fresh-and-github-rest-api-32h6.md]: Thanks  @thegaryroberts , I added standard HTTP status codes.

File: examples/blog/components/BlogAuthor.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export default function BlogAuthor({
+  authorName,
+  authorAvatar,
+}: {
+  authorAvatar: string;
+  authorName: string;
+}) {
+  return (
+    <div className={tw`flex items-center space-x-3`}>
```
Comment (thegaryroberts) [inline examples/blog/components/BlogAuthor.tsx]: In Preact `class=` is preferred over `className=` (although both work). However `class` is being used on the Fresh documentation Web site so is probably preferred here too.

File: examples/blog/routes/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import { fetchRepoMetadata } from "utils/github.ts";
+import GithubRepo from "components/GithubRepo.tsx";
+import Tag from "../components/Tag.tsx";
+import BlogAuthor from "../components/BlogAuthor.tsx";
+
+export const handler: Handlers = {
+  async GET(req, ctx) {
+    try {
+      const BLOG_DIR = "blog";
+      const slug = ctx.params.slug;
```
Comment (thegaryroberts) [inline examples/blog/routes/[slug].tsx]: You have the option of using destructuring here to get rid of some redundancy:
`const { slug } = ctx.params`

File: examples/blog/routes/[slug].tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      }
+
+      return ctx.render({
+        post: { ...markup.meta, content: markup.content },
+        github,
+      });
+    } catch (error) {
+      // redirect to home page if post not found
+      return new Response(undefined, {
+        status: 302,
```
Comment (thegaryroberts) [inline examples/blog/routes/[slug].tsx]: A status code missed from earlier refactor.

File: examples/blog/routes/[slug].tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        headers: { location: "/" },
+      });
+    }
+  },
+};
+
+export default function Blog({ data }: PageProps) {
```
Comment (thegaryroberts) [inline examples/blog/routes/[slug].tsx]: You could use the generic version `PageProps<T>` to add type for your `data` object to keep later code clear of annotations, e.g. `PageProps<{ post: PostDetail, github: any }>` (used `any` for the `github` as this example codebase has no type/interface available)

**Review by moonstripe — CHANGES_REQUESTED** [review]
For the sake of completeness, add the AuthorTypes from examples/blog/components/BlogAuthor.tsx to examples/blog/types/author.ts

File: examples/blog/routes/[slug].tsx (hunk 3 above)
Comment (harshmangalam) [reply examples/blog/routes/[slug].tsx:17]: Updated

**Comment by harshmangalam** [comment]
> For the sake of completeness, add the AuthorTypes from examples/blog/components/BlogAuthor.tsx to examples/blog/types/author.ts

Added

**Pull request closed without merging**